File: Scripts/data_filtering.R
## ----------------------------------------------------------------------------
## Script name: combine_data.R
##
## Purpose: This script filters motifs based on FIMO results
##
## Author: Yaprak Yigit
##
## Date Created: 02-05-2023
##
## Email: y.yigit@st.hanze.nl
## ----------------------------------------------------------------------------
output_file <- "/home/ubuntu/Yaprak/cnn_dataset.csv"

c_difficile <- read.table("/home/ubuntu/Yaprak/FIMO/EXPREG_00000d10/fimo.tsv", sep="\t", header=TRUE)
l_lactis <- read.table("/home/ubuntu/Yaprak/FIMO/EXPREG_00001520/fimo.tsv", sep="\t", header=TRUE)
motif_dataset <- read.table("/home/ubuntu/Yaprak/motif_dataset.csv", sep =",", header=TRUE)

# Find the overlapping motifs and remove them from dataframe l_lactis
overlap <- intersect(l_lactis$sequence_name, c_difficile$sequence_name)
l_lactis <- l_lactis[!l_lactis %in% overlap]

checked_motifs <- rbind(c_difficile, l_lactis)
cnn_dataset <- motif_dataset[motif_dataset$index %in% checked_motifs$sequence_name,]
write.csv(cnn_dataset, output_file)

File: Scripts/data_filtering.py
#!usr/bin/env python3

"""
data_filtering.py

Filters FIMO output based on regulon genes and stores the resulting output in a fasta file
"""

__author__ = "Yaprak Yigit"
__version__ = "0.1"

import os
import argparse
import pandas as pd

# Parse command line arguments
parser = argparse.ArgumentParser(description='Combine FIMO motif files and output as a fasta file')
parser.add_argument('fimo_dir', type=str, help='Path to the FIMO directory')
parser.add_argument('output_file', type=str, help='Output file in fasta format')
args = parser.parse_args()

# Read in data
motifs = []
for root, dirs, files in os.walk(args.fimo_dir):
    for file in files:
        if file.endswith(".tsv"):
            motifs.append(pd.read_csv(os.path.join(root, file), sep="\t", header=0))

# Get overlapping sequence names
overlap = set(motifs[0]['sequence_name'])
for i in range(1, len(motifs)):
    overlap &= set(motifs[i]['sequence_name'])

# Combine all motifs and remove overlapping motifs
all_motifs = pd.concat(motifs, ignore_index=True)
all_motifs = all_motifs[~all_motifs['sequence_name'].isin(overlap)]

# Write the remaining motifs to a fasta file
with open(args.output_file, 'w') as open_file:
    for index, row in all_motifs.iterrows():
        open_file.write(f">{row['sequence_name']}\n{row['matched_sequence']}\n")